src/compiler/scala/tools/nsc/typechecker/ContextErrors.scala

```diff
@@ -750,7 +750,12 @@ trait ContextErrors {
 
       // def stabilize
       def NotAValueError(tree: Tree, sym: Symbol) = {
-        issueNormalTypeError(tree, sym.kindString + " " + sym.fullName + " is not a value")
+        /* Give a better error message for `val thread = java.lang.Thread`. */
+        val betterKindString =
+          if (sym.isJavaDefined && sym.isTrait) "Java interface"
+          else if (sym.isJavaDefined && (sym.isClass || sym.isModule)) "Java class"
```

Comment:
Can Java Enums / Annotations arrive here? They could warrant custom descriptions.

"Non-Scala" would really be more accurate than "Java", but that doesn't really roll off the tongue.

Reply:
I suppose we could just say "class/interface/enum/annotation java.util.Xxx is not a value", and leave off the "Java " part.

Reply:
Alright. I tried a change (again) to `symbolKind` and here are the results:

https://github.com/scala/scala/compare/2.12.x...hrhino:java-kinds-2

I'm okay with this, but wdyt?

Reply:
So, I've come up with an even more extensive list of `symbolKind` strings than that, and it turns out that some occurrences of `object` come from elsewhere (`Type#objectString` being the most recent one I found).

I'm sure I'll miss some. Do you think that the potential inconsistency from places I've missed is too likely to cause confusion worse than this patch is meant to fix?

Reply:
I think not, and we can fix them in the future. The change here looks good to me, and I'm happy to also include the other commit from your branch linked above.